shell/platform/android/io/flutter/embedding/engine/dynamicfeatures/DynamicFeatureManager.java

```diff
@@ -85,7 +100,36 @@
    *     associated Dart deferred library, loading unit id should a negative value and moduleName
    *     must be non-null.
    */
-  public abstract void downloadDynamicFeature(int loadingUnitId, String moduleName);
+  public abstract void installDynamicFeature(int loadingUnitId, String moduleName);
+
+  /**
+   * Gets the current state of the installation session corresponding to the specified loadingUnitId
+   * and/or moduleName.
+   *
+   * <p>Invocations of {@link installDynamicFeature} typically result in asynchronous downloading
```

Comment:
though you can check the completion of that call by just waiting for that future itself no? Explain why they need this

Reply:
This is for displaying things like loading bars, status messages, etc. The status here is purely informational and is not necessary for function.

Reply:
I thought you added somewhere the redundancy with `loadLibrary`. Best be explicit about the web that exists everywhere. i.e. if you installDynamicFeature, you don't really have to check this or listen or the channel, if you listen on the channel, you don't really have to wait for the loadLibrary future, if you loadLibrary, it's the same as installDynamicFeature with loading unit etc, and fully create the web and cross reference everything to everything. 

i.e. we don't want users guessing there's 3-4 APIs that can more or less do the same thing. What am I missing out if I just use 1 and don't use all of them.